packages/@n8n/db/src/repositories/workflow-statistics.repository.ts

```diff
@@ -55,59 +55,59 @@ export class WorkflowStatisticsRepository extends Repository<WorkflowStatistics>
 		isRootExecution: boolean,
 	): Promise<StatisticsUpsertResult> {
 		const dbType = this.globalConfig.database.type;
-		const { tableName } = this.metadata;
+		const escapedTableName = this.manager.connection.driver.escape(this.metadata.tableName);
+
 		try {
 			if (dbType === 'sqlite') {
 				await this.query(
-					`INSERT INTO "${tableName}" ("count", "rootCount", "name", "workflowId", "latestEvent")
-					VALUES (1, ${isRootExecution ? '1' : '0'}, "${eventName}", "${workflowId}", CURRENT_TIMESTAMP)
+					`INSERT INTO ${escapedTableName} ("count", "rootCount", "name", "workflowId", "latestEvent")
+					VALUES (1, ?, ?, ?, CURRENT_TIMESTAMP)
 					ON CONFLICT (workflowId, name)
 					DO UPDATE SET
 						count = count + 1,
 						rootCount = ${isRootExecution ? 'rootCount + 1' : 'rootCount'},
 						latestEvent = CURRENT_TIMESTAMP`,
+					[isRootExecution ? 1 : 0, eventName, workflowId],
 				);
+
 				// SQLite does not offer a reliable way to know whether or not an insert or update happened.
 				// We'll use a naive approach in this case. Query again after and it might cause us to miss the
 				// first production execution sometimes due to concurrency, but it's the only way.
 				const counter = await this.findOne({
 					select: ['count'],
-					where: {
-						name: eventName,
-						workflowId,
-					},
+					where: { name: eventName, workflowId },
 				});
 
 				return (counter?.count ?? 0) > 1 ? 'update' : counter?.count === 1 ? 'insert' : 'failed';
 			} else if (dbType === 'postgresdb') {
 				const upsertRootCount = isRootExecution
-					? `"${tableName}"."rootCount" + 1`
-					: `"${tableName}"."rootCount"`;
+					? `${escapedTableName}."rootCount" + 1`
```

Comment:
Scanners won't pick up on interpolation like this? Guess not if this came up only recently...

Reply:
I parameterized this too. I overlooked this initially.